dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/dubboserver/DubboServer.java

```diff
@@ -0,0 +1,1277 @@
+                    }
+                    if (MetadataService.class.getName().equals(url.getServiceInterface())) {
+                        MetadataUtils.saveMetadataURL(url);
+                    }
+                    Invoker<?> invoker = ServiceConfig.PROXY_FACTORY.getInvoker(serviceConfig.getRef(), (Class) interfaceClass, url);
+                    DelegateProviderMetaDataInvoker wrapperInvoker = new DelegateProviderMetaDataInvoker(invoker, serviceConfig);
+
+                    Exporter<?> exporter = ServiceConfig.PROTOCOL.export(wrapperInvoker);
+                    serviceConfig.addExporter(exporter);
+                }
+
+                MetadataUtils.publishServiceDefinition(url);
+            }
+        }
+        serviceConfig.getExportedUrls().add(url);
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    /**
+     * always export injvm
+     */
+    private void exportLocal(URL url, ServiceConfig serviceConfig) {
+        URL local = URLBuilder.from(url)
+                .setProtocol(LOCAL_PROTOCOL)
+                .setHost(LOCALHOST_VALUE)
+                .setPort(0)
+                .build();
+        Exporter<?> exporter = ServiceConfig.PROTOCOL.export(
+                ServiceConfig.PROXY_FACTORY.getInvoker(serviceConfig.getRef(), (Class) serviceConfig.getInterfaceClass(), local));
+        serviceConfig.addExporter(exporter);
+        logger.info("Export dubbo service " + serviceConfig.getInterfaceClass().getName() + " to local registry url : " + local);
+    }
+
+
+    private void exportServices() {
+        configManager.getServices().forEach(sc -> {
+            if (sc instanceof ServiceConfig) {
+                exportService(sc);
+            }
+        });
+    }
+
+    private void unexportServices() {
+        exportedServices.forEach(sc -> {
+            unexportService(sc);
+        });
+        asyncExportingFutures.clear();
+        exportedServices.clear();
+    }
+
+    private void referServices() {
+        configManager.getReferences().forEach(rc -> {
+            if (rc instanceof ReferenceConfig) {
+                referService(rc);
+            }
+        });
+    }
+
+    private void unreferServices() {
+        if (cache == null) {
+            cache = ReferenceConfigCache.getCache();
+        }
+
+        configManager.getReferences().stream().filter(referenceConfigBase ->
+        {
+            return referenceConfigBase instanceof ReferenceConfig;
+        }).forEach(this::unreferService);
+
+        referenceConfigBase2AsyncExportingFutures.clear();
+        asyncReferringFutures.clear();
+        cache.destroyAll();
+    }
+
+    private void exportService(ServiceConfigBase sc) {
+        if (sc instanceof ServiceConfig) {
+            if (exportAsync) {
+                ExecutorService executor = executorRepository.getServiceExporterExecutor();
+                Future<?> future = executor.submit(() -> {
+                    sc.export();
```

Comment:
Calling `export ServiceConfig(sc)` is more clear, `ServiceConfig Base.export()` is a compatible API